docker/Dockerfile.py

```diff
@@ -0,0 +1,39 @@
+import argparse
+import textwrap
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--role', required=True, choices=('leader', 'worker'))
+parser.add_argument('--sdist', required=True )
+parser.add_argument('--self', required=True )
+options = parser.parse_args()
+
+mesos_role = dict(leader='master', worker='slave')[options.role]
+
+dependencies = ' '.join(['libffi-dev',   # For client side encryption for 'azure' extra with PyNACL
+                         'python-dev',   # For installing Python packages with native code
+                         'python-pip'])  # Bootstrap pip, but needs upgrading, see below
+
+print textwrap.dedent('''
+    FROM mesosphere/mesos-{mesos_role}:1.0.0
+
+    RUN apt-get update && apt-get install -y {dependencies}
+
+    # The stock pip is too old and can't install from sdist with extras
+    RUN pip install --upgrade pip
+
+    # Mesos interface dependency missing on ubuntu
+    RUN pip install protobuf==3.0.0
+
+    COPY {options.sdist} .
+    RUN pip install {options.sdist}[aws,mesos,encryption,cwl]
+
+    # Move the Mesos module onto the python path
+    RUN ln -s /usr/lib/python2.7/site-packages/mesos /usr/local/lib/python2.7/dist-packages/mesos
+
+    # Fix for https://issues.apache.org/jira/browse/MESOS-3793
+    ENV MESOS_LAUNCHER=posix
+
+    # Embed a reference to the image itself so the container can do things like launch another
```

Comment:
Small nit, but this comment reads a bit wonky. I had to read it several times before I understood what it was saying.